src/features/Conversations/ChatsTree.tsx

```diff
@@ -0,0 +1,443 @@
+          return countChildren(children, acc + 1);
+        }
+
+        return acc + 1;
+      }, count);
+
+    const findItem = (items: TreeItem[], itemId: string) =>
+      items.find(({ id }) => id === itemId);
+
+    const buildTree = (flatndItems: FlattenedItem[]): TreeItems => {
+      const root: TreeItem = {
+        id: 'root',
+        children: [],
+        name: '',
+        type: '',
+      };
+      const nodes: Record<string, TreeItem> = { [root.id]: root };
+      const items = flatndItems.map(item => ({ ...item, children: [] }));
+
+      for (const item of items) {
+        const { id, children } = item;
+        const parentId = item.parentId ?? root.id;
+        const parent = nodes[parentId] ?? findItem(items, parentId);
+
+        nodes[id] = { id, children, name: item.name, type: item.type }; //added item and type
+        parent.children.push(item);
+      }
+
+      return root.children;
+    };
+
+    const findItemDeep = (
+      items: TreeItems,
+      itemId: string,
+    ): TreeItem | undefined => {
+      for (const item of items) {
+        const { id, children } = item;
+
+        if (id === itemId) {
+          return item;
+        }
+
+        if (children.length) {
+          const child = findItemDeep(children, itemId);
+
+          if (child) {
+            return child;
+          }
+        }
+      }
+
+      return undefined;
+    };
+
+    const getChildCount = (items: TreeItems, id: string) => {
+      if (!id) {
+        return 0;
+      }
+
+      const item = findItemDeep(items, id);
+
+      return item ? countChildren(item.children) : 0;
+    };
+
+    const handleDragStart = ({ active: { id: currentId } }: DragStartEvent) => {
+      setActiveId(String(currentId));
+      setOverId(String(currentId));
+
+      const currentItem = flattenedItems.find(({ id }) => id === activeId);
+      if (currentItem) {
+        setCurrentPosition({
+          parentId: currentItem.parentId,
+          parentType: currentItem.parentType,
+          overId: String(currentId),
+        });
+      }
+
+      document.body.style.setProperty('cursor', 'grabbing');
+    };
+
+    const handleDragMove = ({ delta }: DragMoveEvent) => {
+      setOffsetLeft(delta.x);
+    };
+
+    const handleDragOver = ({ over }: DragOverEvent) => {
+      setOverId(over?.id?.toString() ?? null);
+    };
+
+    const resetState = () => {
+      setOverId(null);
+      setActiveId(null);
+      setOffsetLeft(0);
+      setCurrentPosition(null);
+
+      document.body.style.setProperty('cursor', '');
+    };
+
+    const handleDragEnd = ({ active, over }: DragEndEvent) => {
+      resetState();
+      if (projected && over) {
+        const { depth, parentId, parentType, currentType } = projected;
+        if (parentType === 'chat') {
+          return;
+        }
+        if (currentType === 'folder' && parentType === 'folder') {
+          return;
+        }
+
+        const clonedItems: FlattenedItem[] = JSON.parse(
```

Comment:
use structuredClone instead of this JSON hack 
https://developer.mozilla.org/en-US/docs/Web/API/structuredClone